testonly/hammer/hammer.go

```diff
@@ -255,13 +261,21 @@ func (s *hammerState) pushSMR(smr *trillian.SignedMapRoot) {
 	for i := smrCount - 1; i > 0; i-- {
 		s.smr[i] = s.smr[i-1]
 	}
+
+	if !s.cfg.CheckSignatures {
+		smr.Signature = nil
+	}
 	s.smr[0] = smr
 }
 
 func (s *hammerState) previousSMR(which int) *trillian.SignedMapRoot {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
-	return s.smr[which]
+	r := s.smr[which]
+	if r != nil && !s.cfg.CheckSignatures {
```

Comment:
Am I right in thinking that this zeroing (and with it the copy of the `SignedMapRoot`) shouldn't be needed (because of the previous change)?

Reply:
It was a bit belt-n-braces.  Maybe it's better as an assertion.

Reply:
I've changed it to an assertion.  See if you like it.

Reply:
Could we reverse the first `if` test and combine with the second?
```go
func (s *hammerState) previousSMR(which int) *trillian.SignedMapRoot {
	s.mu.RLock()
	defer s.mu.RUnlock()
	r := s.smr[which]
	if !s.cfg.CheckSignatures && r != nil && r.Signature != nil {
		panic(fmt.Sprintf("signature should have been cleared before storing SMR %v", r))
	}
	return r
}
```
feels easier to follow (to me at least).

Reply:
Yep, that's better. Done.